### PR TITLE
Added 120FPS and 144FPS option in preferences; added "Fluent" theme; increased tempo limit

### PR DESCRIPTION
**Pull request by chenxi050402**
Added higher FPS options for stabler tempo.
Added a new theme inspired by Microsoft's Fluent Design.
Increased tempo limit because why not

Fixed a bug about the marker not long enough.

### CODE REVIEW

**Comment by Bentroen** [comment]
Hello! Thank you for your PR :)

Although I like the attempt to make the tempo more stable, I'm not too fond of adding higher FPS options. The program already lags enough trying to draw 60 FPS, and although a modern PC could probably handle drawing more frames, I don't think the difference would be too significant. In theory, trying to push out more frames should make the playback smoother, but from my experience, it sort of causes the opposite effect: by forcing the program to draw more frames, you make it lag more, which makes the playback less smooth. Ultimately, assessing whether it'd be a good addition would require a lot of testing on different setups.

On an unrelated note, why is the entire file being changed on the diff? It's making it sort of difficult to track what changed.

**Comment by chenxi050402** [comment]
It still sounds a little "swing" feeling when playing using 60hz and my computer can handle 144hz very well so it's ok to have an option for people with higher standards and better a computer.

File: scripts/draw_window_preferences/draw_window_preferences.gml [hunk 1]
```diff
@@ -1,205 +1,213 @@
...
+			if (autosavemins != a) {changed = 1 tonextsave = autosavemins}
+		}
+		popup_set_window(x1 + 306, y1 + 110, 180, 16, "The amount of minutes between each auto-save.")
+	
+	    draw_areaheader(x1 + 22, y1 + 164, 218, 120, "Theme")
+	    if (draw_radiobox(x1 + 40, y1 + 164 + 16, theme == 0, "Aqua", "Use the aqua theme.")) {theme = 0 change_theme()}
+		if (draw_radiobox(x1 + 40, y1 + 164 + 16 + 20, theme == 2, "Dark", "Use the dark theme.")) {theme = 2 change_theme()}
+	    if (draw_radiobox(x1 + 40, y1 + 164 + 16 + 20 + 20, theme == 1, "90s", "Use the 90s theme.")) {theme = 1 change_theme()}
+		if (draw_checkbox(x1 + 40, y1 + 164 + 16 + 20 + 20 + 25, blackout, "Blackout mode", "Makes the workspace background solid black, so you can\nkey it out in your video editor when recording the screen.")) blackout = !blackout
+	    draw_areaheader(x1+258,y1+164,220,60,"Max frame rate (experimental)")
```
Comment (chenxi050402) [inline scripts/draw_window_preferences/draw_window_preferences.gml:95]: Changed section

File: scripts/draw_window_preferences/draw_window_preferences.gml [hunk 2]
```diff
@@ -1,205 +1,213 @@
...
+	    }
+		if (draw_radiobox(x1+274+64,y1+164+16,refreshrate == 2,"120FPS","Run the program at 120FPS.")) {
+			game_set_speed(120,gamespeed_fps)
+			refreshrate=2
+		}
+		if (draw_radiobox(x1+274+64,y1+164+16+20,refreshrate == 3,"144FPS","Run the program at 144FPS.")) {
+			game_set_speed(144,gamespeed_fps)
+			refreshrate=3
+		}
+
```
Comment (chenxi050402) [inline scripts/draw_window_preferences/draw_window_preferences.gml:112]: End changed section

Comment (chenxi050402) [reply scripts/draw_window_preferences/draw_window_preferences.gml:112]: Sorry I don't use Github a lot :)

**Comment by Bentroen** [comment]
I like the new theme a lot!! Great work :)

We do have a slight problem though. I see you added the Segoe font family to the source; however, Microsoft **does not allow** redistributing Windows fonts as bitmaps inside of a program. From the [Font redistribution FAQ for Windows](https://docs.microsoft.com/en-us/typography/fonts/font-faq):

> **Can I embed the fonts into a game, application or device I’m developing based on the document font embedding permissions?**
>
> No, document font embedding permissions relate to embedding fonts in documents only, not embedding fonts in games, apps and devices.
>
>**If I convert the font into a bitmap font can I include that in my game or app?**
>
> No, converting Windows fonts to other formats does not change the rules around embedding or redistribution, and format conversion itself is not allowed. Many Microsoft supplied fonts are available for app and game licensing through the original font foundry or Monotype.

The program has a history of using various libraries without proper credit (see https://github.com/HielkeMinecraft/OpenNoteBlockStudio/pull/84#issuecomment-542755442), which is something I'm actively trying to revert. Adding unlicensed fonts would be a step in the wrong direction.

Microsoft does provide an open-source alternative to Segoe UI:

> **How about Segoe UI? I love it and would like to include that in my non-Windows app or game. Is it available from Monotype?**
> 
> No, as Segoe UI is both our user interface and corporate branding font, it is not available for use outside of Microsoft products on non-Windows platforms. However, we do have a Segoe-compatible open-source font you can use: [Selawik](https://github.com/Microsoft/Selawik).

This should be the way to go. I won't revert the merge, but we should open another pull request addressing this.

**Comment by Bentroen** [comment]
Some immediate issues I see with the Fluent theme:
- Layer lock icon has volume off icon
- Low contrast between selected color (blue) and icon color (use a softer shade of blue?)
- Low contrast between locked buttons and text (for instance, the lower buttons in instrument settings)
- Last layer is missing an outline
- There doesn't seem to be a pattern between switches and checkboxes on toggles (?)
- Song timestamp bounces around during playback due to the text length changing
- Metronome icon is missing a 1 on the first beat (a limitation on the way it was implemented, which uses the same icon as when stopped)
- Hover and press button bevels are the same color as the background (plain white)

Now onto issues revealed by the 120/144 FPS options:
- Menu opening animations are faster on higher FPS, which makes menus close instantly when clicked
- Edge scrolling when dragging a selection/moving notes is accentuated on higher FPS
- Sound count is updated far too often
- A "114514 FPS" option was forgotten there...? xD

(Most of these actually used to be a problem before this addition, but the higher FPS counts have made them much more apparent.)

Overall, you did an amazing job with these additions. I'm surprised at how good you made the theme look despite all the limitations of the code.

Once these issues are addressed, I'm quite confident that we can make it the default theme! (would be really neat to see a dark version of it as well :)

**Comment by chenxi050402** [comment]
> Some immediate issues I see with the Fluent theme:
> 
> * Layer lock icon has volume off icon
> * Low contrast between selected color (blue) and icon color (use a softer shade of blue?)
> * Low contrast between locked buttons and text (for instance, the lower buttons in instrument settings)
> * Last layer is missing an outline
> * There doesn't seem to be a pattern between switches and checkboxes on toggles (?)
> * Song timestamp bounces around during playback due to the text length changing
> * Metronome icon is missing a 1 on the first beat (a limitation on the way it was implemented, which uses the same icon as when stopped)
> * Hover and press button bevels are the same color as the background (plain white)
> 
> Now onto issues revealed by the 120/144 FPS options:
> 
> * Menu opening animations are faster on higher FPS, which makes menus close instantly when clicked
> * Edge scrolling when dragging a selection/moving notes is accentuated on higher FPS
> * Sound count is updated far too often
> * A "114514 FPS" option was forgotten there...? xD
> 
> (Most of these actually used to be a problem before this addition, but the higher FPS counts have made them much more apparent.)
> 
> Overall, you did an amazing job with these additions. I'm surprised at how good you made the theme look despite all the limitations of the code.
> 
> Once these issues are addressed, I'm quite confident that we can make it the default theme! (would be really neat to see a dark version of it as well :)

* I made lock layer a volume off because locking a layer functions as muting a layer
* I accidentally made it blue (it was supposed to be grey)
* Working on
* I removed the outline because there isn't supposed to be an outline in Fluent Design if we're talking about the same outline
* There is no pattern in my windows settings and powertoys preferences of checkboxes and flip switches either
* It will be better when positioning using left coords
* I made the timer icon follow Windows's file name logic
* Forgot to add that
* It's noticeable even on 60fps just not significant, I hold the tab and move to menu items and release to go to settings
* I think it works like that in Minecraft too
* It's supposed to be a Unlimited FPS option, changing...

**Pull request merged**